common/ManagementTestShared/ManagementRecordTestBase.cs

```diff
@@ -0,0 +1,181 @@
+            await StopTestRecording();
+            await CleanupResourceGroupsAsync();
+            Logger?.Dispose();
+            Logger = null;
+            await OnOneTimeTearDownAsync();
+        }
+        /// <summary>
+        /// A method called by NUnit framework to setup recordings for new test and run onsetup
+        /// </summary>
+        [SetUp]
+        protected async Task StartTestRecording()
+        {
+            await StopTestRecording();
+            TestContext.TestAdapter test = TestContext.CurrentContext.Test;
+            if (Mode != RecordedTestMode.Live &&
+                test.Properties.ContainsKey("SkipRecordings"))
+            {
+                throw new IgnoreException((string)test.Properties.Get("SkipRecordings"));
+            }
+            Recording = new TestRecording(Mode, GetSessionFilePath(), Sanitizer, Matcher);
+            TestEnvironment.Mode = Mode;
+            TestEnvironment.SetRecording(Recording);
+            InitializeClients();
+            await OnSetupAsync();
+        }
+        /// <summary>
+        /// Called by NUnit upon completion of test and saves the recording if running in record mode
+        /// </summary>
+        [TearDown]
+        protected async Task StopTestRecording()
+        {
+            bool save = TestContext.CurrentContext.Result.FailCount == 0;
+#if DEBUG
+            save |= SaveDebugRecordingsOnFailure;
+#endif
+            Recording?.Dispose(save);
+            await OnTearDownAsync();
+        }
+
+        /// <summary>
+        /// Create a resource group to be monitored by the test framework.
+        /// </summary>
+        protected ResourcesManagementClient GetResourceManagementClient()
+        {
+            var options = InstrumentClientOptions(new ResourcesManagementClientOptions());
+            options.AddPolicy(CleanupPolicy, HttpPipelinePosition.PerCall);
+            return GetManagementClient<ResourcesManagementClient>(options);
+        }
+
+        /// <summary>
+        /// Delete all monitored resource groups.
+        /// </summary>
+        protected async Task CleanupResourceGroupsAsync()
+        {
+            if (CleanupPolicy != null && Mode != RecordedTestMode.Playback)
+            {
+                var resourceGroupsClient = new ResourcesManagementClient(
+                    TestEnvironment.SubscriptionId,
+                    TestEnvironment.Credential,
+                    new ResourcesManagementClientOptions()).ResourceGroups;
+                foreach (var resourceGroup in CleanupPolicy.ResourceGroupsCreated)
+                {
+                    await resourceGroupsClient.StartDeleteAsync(resourceGroup);
+                }
+                CleanupPolicy.ResourceGroupsCreated.Clear();
+            }
+        }
+        /// <summary>
+        /// Get a management client with the test enviroment subscription
+        /// </summary>
+        protected T GetManagementClient<T>(ClientOptions options) where T : class
```

Comment:
Yes because the recording get's disposed off and then we need to re-instrument the client with the new recording for the new test. 

Given the current recording class, which we have not touched, we either have to do this, or everything would be shoved in one massive recording. 

As a longer term task we could change this by modifying the TestRecording class so we don't have to do this. Say, just call save recording on the class then and clear everything.

Reply:
I think you explained why we need to call this, but not necessarily why it needs to be called WITHOUT calling line 149.